File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
class Solution {
    public String solution(String[] survey, int[] choices) {
        
        // 1 = R , T || 2 = C , F || 3 = J , M || 4 = A , N
        // 선택지 = 비동의 <- 1 2 3 || 4 || 5 6 7 -> 동의
        
        String answer = "";
        int[] result = new int[8];
        String[] str = {"R","T","C","F","J","M","A","N"};

        for(int i = 0; i < survey.length; i++){
            String category = survey[i];
            int select = choices[i];

            if(category.equals("RT")){
                result[0] += select < 4 ? 4 - select : 0;
                result[1] += select > 4 ? select - 4 : 0;
            }else if(category.equals("TR")){
                result[1] += select < 4 ? 4 - select : 0;
                result[0] += select > 4 ? select - 4 : 0;
            }else if(category.equals("CF")){
                result[2] += select < 4 ? 4 - select : 0;
                result[3] += select > 4 ? select - 4 : 0;
            }else if(category.equals("FC")){
                result[3] += select < 4 ? 4 - select : 0;
                result[2] += select > 4 ? select - 4 : 0;
            }else if(category.equals("JM")){
                result[4] += select < 4 ? 4 - select : 0;
                result[5] += select > 4 ? select - 4 : 0;
            }else if(category.equals("MJ")){
                result[5] += select < 4 ? 4 - select : 0;
                result[4] += select > 4 ? select - 4 : 0;
            }else if(category.equals("AN")){
                result[6] += select < 4 ? 4 - select : 0;
                result[7] += select > 4 ? select - 4 : 0;
            }else if(category.equals("NA")){
                result[7] += select < 4 ? 4 - select : 0;
                result[6] += select > 4 ? select - 4 : 0;
            }
        }

        for(int j = 0; j < 8; j += 2){
            if(result[j] >= result[j + 1]){
                answer += str[j];
            }else if(result[j + 1] > result[j]){
                answer += str[j + 1];
            }
        }
        
        
        return answer;
    }
    
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// RT CF JM AN

char* solution(const char* survey[], size_t survey_len, int choices[], size_t choices_len) {
    char* answer = (char*)malloc(4 * sizeof(char));
    char* mbti = "RTCFJMAN";
    int category[8] = {0,};
    
    for(int i = 0; i < choices_len; i++){
        for(int j = 0; j < 8; j++){
            if(survey[i][1] == mbti[j]){
                category[j] += choices[i] - 4;
            }
        }
    }
    
    for(int k = 0; k < 8; k += 2){
        if(category[k] >= category[k + 1]){
            answer[k / 2] = mbti[k];
        }else {
            answer[k / 2] = mbti[k + 1];
        }
    }
    
    answer[4] = '\0';
    
    return answer;
}